src/wp-includes/block-template-utils.php

```diff
@@ -1,23 +1,481 @@
+		),
+		'archive'        => array(
+			'title'       => _x( 'Archive', 'Template name' ),
+			'description' => __( 'The archive template displays multiple entries at once. It is used as a fallback for the Category, Author, and Date templates, which take precedence when they are available.' ),
+		),
+		'author'         => array(
+			'title'       => _x( 'Author', 'Template name' ),
+			'description' => __( 'Archive template used to display a list of posts from a single author.' ),
+		),
+		'category'       => array(
+			'title'       => _x( 'Category', 'Template name' ),
+			'description' => __( 'Archive template used to display a list of posts from the same category.' ),
+		),
+		'taxonomy'       => array(
+			'title'       => _x( 'Taxonomy', 'Template name' ),
+			'description' => __( 'Archive template used to display a list of posts from the same taxonomy.' ),
+		),
+		'date'           => array(
+			'title'       => _x( 'Date', 'Template name' ),
+			'description' => __( 'Archive template used to display a list of posts from a specific date.' ),
+		),
+		'tag'            => array(
+			'title'       => _x( 'Tag', 'Template name' ),
+			'description' => __( 'Archive template used to display a list of posts with a given tag.' ),
+		),
+		'attachment'     => array(
+			'title'       => __( 'Media' ),
+			'description' => __( 'Template used to display individual media items or attachments.' ),
+		),
+		'search'         => array(
+			'title'       => _x( 'Search', 'Template name' ),
+			'description' => __( 'Template used to display search results.' ),
+		),
+		'privacy-policy' => array(
+			'title'       => __( 'Privacy Policy' ),
+			'description' => '',
+		),
+		'404'            => array(
+			'title'       => _x( '404', 'Template name' ),
+			'description' => __( 'Template shown when no content is found.' ),
+		),
+	);
+
+	/**
+	 * Filters the list of template types.
+	 *
+	 * @param array $default_template_types An array of template types, formatted as [ slug => [ title, description ] ].
+	 *
+	 * @since 5.x.x
+	 */
+	return apply_filters( 'default_template_types', $default_template_types );
+}
+
+/**
+ * Checks whether the input 'area' is a supported value.
+ * Returns the input if supported, otherwise returns the 'uncategorized' value.
+ *
+ * @access private
+ * @since 5.9.0
+ *
+ * @param string $type Template part area name.
+ *
+ * @return string Input if supported, else the uncategorized value.
+ */
+function _filter_block_template_part_area( $type ) {
+	$allowed_areas = array_map(
+		function ( $item ) {
+			return $item['area'];
+		},
+		get_allowed_block_template_part_areas()
+	);
+	if ( in_array( $type, $allowed_areas, true ) ) {
+		return $type;
+	}
+
+	/* translators: %1$s: Template area type, %2$s: the uncategorized template area value. */
+	$warning_message = sprintf( __( '"%1$s" is not a supported wp_template_part area value and has been added as "%2$s".' ), $type, WP_TEMPLATE_PART_AREA_UNCATEGORIZED );
```

Comment:
During the 5.9.0 cycle, we've been moving the translators comment directly above it's function and putting longer lines of code like this on multiple lines (for improved readability):

```suggestion
	$warning_message = sprintf(
		/* translators: %1$s: Template area type, %2$s: the uncategorized template area value. */
		__( '"%1$s" is not a supported wp_template_part area value and has been added as "%2$s".' ),
		$type,
		WP_TEMPLATE_PART_AREA_UNCATEGORIZED
	);
```

Also `wp_template_part` in its current form could confuse translators. Consider a non-code version or adding another placeholder.